consensus/bor/rest.go

```diff
@@ -97,12 +100,17 @@ func (h *HeimdallClient) FetchWithRetry(rawPath string, rawQuery string) (*Respo
 	u.RawQuery = rawQuery
 
 	for {
-		res, err := h.internalFetch(u)
-		if err == nil && res != nil {
-			return res, nil
+		select {
+		case <-h.closeCh:
```

Comment:
The fix looks fine. How do you thing does it worth it to add a test with something like https://github.com/uber-go/goleak ?

Reply:
Converting this to a separate task, as per our discussion.